core/src/rpc.rs

```diff
@@ -330,7 +331,11 @@ impl JsonRpcRequestProcessor {
         slot: Slot,
         encoding: Option<RpcTransactionEncoding>,
     ) -> Result<Option<RpcConfirmedBlock>> {
-        Ok(self.blockstore.get_confirmed_block(slot, encoding).ok())
+        if self.config.enable_get_confirmed_block {
+            Ok(self.blockstore.get_confirmed_block(slot, encoding).ok())
+        } else {
+            Ok(None)
+        }
```

Comment:
I went for the approach of completely disabling `getConfirmedBlock` instead of returning partial results if the transaction status and rewards recorder services are disabled, as this seemed likely to be an easier debug when the first couple people wonder why their node is not returning the full data set